wsagent/che-core-git-impl-jgit/src/main/java/org/eclipse/che/git/impl/jgit/JGitConnection.java

```diff
@@ -528,50 +530,86 @@ protected void onEndTask(String taskName, int workCurr, int workTotal, int perce
+            staged.addAll(status.getAdded());
+            staged.addAll(status.getChanged());
+            staged.addAll(status.getRemoved());
+            List<String> changed = new ArrayList<>(staged);
+            changed.addAll(status.getModified());
+            changed.addAll(status.getMissing());
+            List<String> specifiedChanged = specified.stream()
+                                                     .filter(path -> changed.stream().anyMatch(c -> c.startsWith(path)))
+                                                     .collect(Collectors.toList());
+
+            // Check that there are changes present for commit, if 'isAmend' is disabled
+            if (!params.isAmend()) {
+                // Check that there are staged changes present for commit, or any changes if 'isAll' is enabled
+                if (status.isClean()) {
+                    throw new GitException("Nothing to commit, working directory clean");
+                } else if (!params.isAll() && (specified.isEmpty() ? staged.isEmpty() : specifiedChanged.isEmpty())) {
+                    throw new GitException("No changes added to commit");
+                }
+            /*
+            By default Jgit doesn't allow to commit not changed specified paths. According to setAllowEmpty method documentation,
+            setting this flag to true must allow such commit, but it won't because Jgit has a bug:
+            https://bugs.eclipse.org/bugs/show_bug.cgi?id=510685. As a workaround, specified paths of the commit command will contain
+            only changed and specified paths. If other changes are present, but the list of changed and specified paths is empty,
+            throw exception to prevent committing other paths. TODO Remove this check when the bug will be fixed.
+            */
+            } else if (!specified.isEmpty() && !(params.isAll() ? changed.isEmpty() : staged.isEmpty()) && specifiedChanged.isEmpty()) {
+                throw new GitException(format("Changes are present but not changed path%s specified for commit.",
+                                              specified.size() > 1 ? "s were" : " was"));
             }
 
             CommitCommand commitCommand = getGit().commit()
-                                                  .setCommitter(committerName, committerEmail).setAuthor(committerName, committerEmail)
+                                                  .setCommitter(committerName, committerEmail)
+                                                  .setAuthor(committerName, committerEmail)
                                                   .setMessage(message)
                                                   .setAll(params.isAll())
-                                                  .setAmend(params.isAmend());
+                                                  .setAmend(params.isAmend())
+                                                  // Allow empty commits with specified paths with amend
```

Comment:
comment says about commits with specified paths but code in next line doesn't do it. Fix code or comment